src/providers/swift.rs

```diff
@@ -0,0 +1,175 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::{bail, Result};
+
+static DEFAULT_SWIFT_VERSION: &str = "5.5.3";
+
+// From: https://lazamar.co.uk/nix-versions/?channel=nixpkgs-unstable&package=swift
+static AVAILABLE_SWIFT_VERSIONS: &[(&str, &str)] = &[
+    ("3.1", "aeaa79dc82980869a88a5955ea3cd3e1944b7d80"),
+    ("3.1.1", "8414d8386b9a6b855b291fb3f01a4e3b04c08bbb"),
+    ("4.0.3", "2c9d2d65266c2c3aca1e4c80215de8bee5295b04"),
+    ("4.1", "92a047a6c4d46a222e9c323ea85882d0a7a13af8"),
+    ("4.1.3", "a3962299f14944a0e9ccf8fd84bd7be524b74cd6"),
+    ("4.2.1", "7ff8a16f0726342f0a25697867d8c1306d4da7b0"),
+    ("4.2.3", "3fa154fd7fed3d6a94322bf08a6def47d6f8e0f6"),
+    ("5.0.1", "4599f2bb9a5a6b1482e72521ead95cb24e0aa819"),
+    ("5.0.2", "a9eb3eed170fa916e0a8364e5227ee661af76fde"),
+    ("5.1.1", "9986226d5182c368b7be1db1ab2f7488508b5a87"),
+    ("5.4", "c82b46413401efa740a0b994f52e9903a4f6dcd5"),
+    ("5.4.2", "c82b46413401efa740a0b994f52e9903a4f6dcd5"),
+    ("5.5.2", "6d02a514db95d3179f001a5a204595f17b89cb32"),
+    ("5.5.3", "bf972dc380f36a3bf83db052380e55f0eaa7dcb6"),
+];
+
+pub struct SwiftProvider {}
+
+impl Provider for SwiftProvider {
+    fn name(&self) -> &str {
+        "swift"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Package.swift")
+            || (app.includes_file("Package.swift") && app.includes_file("Package.resolved")))
```

Comment:
The second half of this `||` is redundant since it will already have matched if `Package.swift` exists.